standards/core/peer-exchange.md

```diff
@@ -66,26 +66,40 @@ message PeerInfo {
   bytes enr = 1;
 }
 
-message PeerExchangeQuery {
+message PeerExchangeRequest {
   uint64 num_peers = 1;
 }
 
 message PeerExchangeResponse {
   repeated PeerInfo peer_infos = 1;
+  uint32 status = 10;
+  optional string desc = 11;
 }
 
 message PeerExchangeRPC {
-  PeerExchangeQuery query = 1;
+  PeerExchangeRequest request = 1;
   PeerExchangeResponse response = 2;
 }
 
 ```
 
 The `enr` field contains a Waku ENR as specified in [WAKU2-ENR](./enr.md).
 
-Requesters send a `PeerExchangeQuery` to a peer.
+Requesters send a `PeerExchangeRequest` to a peer.
 Responders SHOULD include a maximum of `num_peers` `PeerInfo` instances into a response.
 Responders send a `PeerExchangeResponse` to requesters containing a list of `PeerInfo` instances, which in turn hold an ENR.
+Responders, MUST include a `status` in the response in any case and possible descriptive form of the problem if any.
```

Comment:
should this be change it to SHOULD so that clients can function with both new and older model where codes are not returned?

Reply:
Well, I'm not pretty sure. As - for me - a spec describes the state we are aiming and not the past we are coming from. So I would expect any implementation of the protocol to behave as "MUST" and not "SHOULD" or "MIGHT".
Of course for backward compatibility according to protobuf 3 specs these fields will get a default value according to their types... I believe this case `status` will take `0` and will mean UNKNOWN. So either old requester or responder will be able to operate. But sooner or later I expect them to follow the spec.
@chaitanyaprem , @jm-clius : WDYT?